optimizely/client/client.go

```diff
@@ -133,6 +135,115 @@ func (o *OptimizelyClient) GetEnabledFeatures(userContext entities.UserContext)
 	return enabledFeatures, nil
 }
 
+// GetFeatureVariableString returns string feature variable value
+func (o *OptimizelyClient) GetFeatureVariableString(featureKey string, variableKey string, userContext entities.UserContext) (value string, err error) {
+	val, err := o.getFeatureVariable("", featureKey, variableKey, userContext)
+	if err == nil {
+		switch val.(type) {
+		case string:
+			return val.(string), err
+		default:
+			break
+		}
+	}
+	return "", err
+}
+
+func (o *OptimizelyClient) getFeatureVariable(valueType interface{}, featureKey string, variableKey string, userContext entities.UserContext) (value interface{}, err error) {
+	if !o.isValid {
+		errorMessage := "Optimizely instance is not valid. Failing getFeatureVariable."
+		err := errors.New(errorMessage)
+		logger.Error(errorMessage, nil)
+		return nil, err
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			errorMessage := fmt.Sprintf(`Optimizely SDK is panicking with the error "%s"`, string(debug.Stack()))
+			err = errors.New(errorMessage)
+			logger.Error(errorMessage, err)
+		}
+	}()
+
+	projectConfig := o.configManager.GetConfig()
+
+	if reflect.ValueOf(projectConfig).IsNil() {
+		return nil, fmt.Errorf("project config is null")
+	}
+
+	featureFlag, err := projectConfig.GetFeatureFlagByKey(featureKey)
+	if err != nil {
+		logger.Error("Error retrieving feature flag", err)
+		return nil, err
+	}
+	variable, err := featureFlag.GetVariable(variableKey)
+	if err != nil {
+		logger.Error("Error retrieving variable", err)
+		return nil, err
+	}
+
+	var featureValue = variable.DefaultValue
+
+	feature, err := projectConfig.GetFeatureByKey(featureKey)
+	if err != nil {
+		logger.Error("Error retrieving feature", err)
+		return nil, err
+	}
+	featureDecisionContext := decision.FeatureDecisionContext{
+		Feature:       &feature,
+		ProjectConfig: projectConfig,
+	}
+
+	featureDecision, err := o.decisionService.GetFeatureDecision(featureDecisionContext, userContext)
+	if err == nil {
+		for _, v := range featureDecision.Variation.Variables {
+			if v.ID == variable.ID && featureDecision.Variation.FeatureEnabled {
+				featureValue = v.Value
+				break
+			}
+		}
+	}
+
+	var typeName = ""
+	var valueParsed interface{}
+	switch valueType.(type) {
```

Comment:
Instead of passing in a variable for which we then get the type of, we can just pass in a string with the name of the type. This is less expensive to do.